File: aml-backend/src/main/java/dev/abarmin/aml/dashboard/block/social/SocialNetworksBlockProps.java
package dev.abarmin.aml.dashboard.block.social;

import dev.abarmin.aml.dashboard.domain.BlockProps;
import jakarta.validation.Valid;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public final class SocialNetworksBlockProps implements BlockProps {
  @Valid
  @Builder.Default
  private List<SocialNetworkLink> links = new ArrayList<>();

}

File: aml-frontend-private/src/model/SocialNetworksBlockProps.ts
import type { BlockProps } from "./BlockProps";

export interface SocialNetworksBlockProps extends BlockProps {
  links: SocialNetworkLink[];
}

export type SocialNetwork = "FACEBOOK" | "TWITTER" | "X" | "INSTAGRAM";

export interface SocialNetworkLink {
  url: string;
  network: SocialNetwork;
}
